content/news/2023-10-21-bevy-0.12/index.md

```diff
@@ -19,6 +19,319 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
+
+* Pipeline
+* BindGroup or its corresponding dynamic offsets
+* Index/vertex buffer
+* Index/vertex range
+* Instance range
+
+Pipelines usually vary due to using different shaders in custom materials, or using variants of a material due to shader defs as the shader defs produce different shaders. Bind group bindings can change due to different material textures, different buffers, or needing to bind different parts of some buffers using dynamic offsets. Index/vertex buffers and/or ranges change per mesh asset. Instance range is what we want to leverage for instanced draws.
+
+#### Assumptions
+
+The design of the automatic batching/instanced draws in Bevy makes some assumptions to enable a reasonable solution:
+
+* Only entities with prepared assets are queued to render phases
+* View bindings are constant across a render phase for a given draw function, as phases are per-view
+* `batch_and_prepare_render_phase` is the only system that performs batching and has sole responsibility for preparing per-instance (i.e. mesh uniform) data
+
+If these assumptions do not work for your use case, then you can add the `NoAutomaticBatching` component to your entities to opt-out and do your own thing. Note that mesh uniform data will still be written to the `GpuArrayBuffer` and can be used in your own mesh bind groups.
+
+#### Instanced Draw Performance
+
+We can batch draws into a single instanced draw in some situations now that per-instance mesh uniform data is in a `GpuArrayBuffer`. If the mesh entity is using the same mesh asset, and same material asset, then it can be batched!
+
+Using the same approach as 0.11 with one dynamic offset binding per mesh entity, and comparing to either storage buffers or batched uniform buffers:
+
+2D meshes: `bevymark --benchmark --waves 160 --per-wave 1000 --mode mesh2d --ordered-z` which spawns 160 waves of 1000 2D quad meshes, producing 160 instanced draws of 1000 instances per draw.
+
+![0.12-2DMeshes](0.12-2DMeshes.svg)
+<div style="font-size: 1.0rem" class="release-feature-authors">Tested on an M1 Max, limiting the BatchedUniformBuffer batch size to 1 versus how it works in 0.12.</div>
+
+3D meshes: `many_cubes` which spawn 160,000 cubes, of which ~11,700 are visible in the view. These are drawn using a single instanced draw of all visible cubes which enables up to **100% increase in frame rate (2x)**!
+
+![0.12-3DMeshes](0.12-3DMeshes.svg)
+<div style="font-size: 1.0rem" class="release-feature-authors">Tested on an M1 Max, limiting the BatchedUniformBuffer batch size to 1 versus how it works in 0.12.</div>
+
+These performance benefits can be leveraged on all platforms, including WebGL2!
+
+#### What is next for batching/instancing and beyond?
```

Comment:
This section is full of too much context. This should be reduced to a few short / one sentence bullet points and moved to the generic "whats next" section at the end.